### PR TITLE
Use Terra's Config Modules

### PR DESCRIPTION
**Pull request by emilyrohrbough**
### Summary
This PR is a WIP
- Use browerslist-config-terra, eslint-config-terra, and stylelint-config-terra modules.
- Fix lint errors.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
   },
   "repository": {
     "type": "git",
-    "url": "git+https://github.com/cerner/terra-framework.git"
+    "url": "githttps://github.com/cerner/terra-framework.git"
```
Comment (bjankord) [inline package.json]: why the removal of the `+` character here?

File: package.json [hunk 2]
```diff
@@ -21,6 +21,15 @@
   "bugs": {
     "url": "https://github.com/cerner/terra-framework/issues"
   },
+  "browserslist": [
+    "extends browserslist-config-terra"
+  ],
+  "eslintConfig": {
+    "extends": "terra"
```
Comment (tomleewu) [inline package.json]: Should this be
```
"eslintConfig": {
  "extends": "eslint-config-terra"
},
```

Comment (emilyrohrbough) [reply package.json]: Good question! Either definition is valid, but eslint allows extends to obmit the `estlint-config` prefix. Here is the link to the docs: https://eslint.org/docs/user-guide/configuring#using-a-shareable-configuration-package.

Comment (bjankord) [reply package.json]: I feel like it should, I'm surprised using just `terra` works, though thats also what we use in terra-core. @emilyrohrbough what your thoughts on this?

Comment (emilyrohrbough) [reply package.json]: Eslint doesn't require the prefix to be included to successfully load the configuration. Eslint searched for the module `eslint-config-terra` when given `terra`. I am unopinionated to how we define it in the configuration.

Comment (bjankord) [reply package.json]: Let's keep it as is, TIL something new about ESLint configs 👍

File: packages/terra-application-layout/src/menu/RoutingMenu.jsx [hunk 3]
```diff
@@ -58,22 +58,6 @@ class RoutingMenu extends React.Component {
     }));
   }
 
-  /**
```
Comment (emilyrohrbough) [inline packages/terra-application-layout/src/menu/RoutingMenu.jsx]: Change due to `react/sort-comp` ordering rules

File: packages/terra-popup/src/_PopupContent.jsx [hunk 4]
```diff
@@ -95,30 +95,6 @@ const defaultProps = {
 };
 
 class PopupContent extends React.Component {
-  static getDimensionStyle(value, maxValue, isAutomatic) {
```
Comment (emilyrohrbough) [inline packages/terra-popup/src/_PopupContent.jsx]: Change due to `react/sort-comp` ordering rules

**Pull request merged**